spec/parsers/generic_spec.rb

```diff
@@ -21,4 +21,10 @@
   it 'matches valid manifest filepaths' do
     expect(described_class.match?('dependencies.csv')).to be_truthy
   end
+
+  it 'raises an error for blank required fields' do
+    csv = %Q!platform,name,requirement\ngo,something,!
+    result = described_class.analyse_contents('dependencies.csv', csv)
+    expect(result[:error_message]).to eq("dependencies.csv: missing field 'requirement' on line 1")
```

Comment:
should be line 2 shouldn't it?

Reply:
Ah yeah, it looks like the CSV library counts the headers as line 0, so it makes sense that we would too (I assumed it might be 1-index-based).

Reply:
WAIT, I'm confused, if I set `headers: true` then it is 1-based:

```
> CSV.parse("my-header\n\"unclosed-row", field_size_limit: 1, headers: true)[0]
CSV::MalformedCSVError: Unclosed quoted field in line 2.
```

Reply:
no memory here, do what makes sense to you 🙂

Reply:
Here's more evidence that the CSV library uses 1-based-index, so I'll go ahead and tweak this logic:

https://ruby-doc.org/stdlib-3.1.1/libdoc/csv/rdoc/CSV.html#method-i-line

Reply:
I just was thinking from the perspective of a user. It doesn't matter what we do back here but if we're displaying an error to the user, I don't think they'll think of the header line as line 0.